src/main/java/nl/ai42/controllers/LoginAndRegisterController.java

```diff
@@ -17,84 +19,124 @@ public class LoginAndRegisterController {
+            invalid_signup_credentials.setText("");
+
+            if (login_username_text_field.getText().isBlank()) 
+                login_username_text_field.setStyle(errorStyle);
+                
+            if (login_password_password_field.getText().isBlank())
+                login_password_password_field.setStyle(errorStyle);
         } else {
-            invalidLoginCredentials.setText("Login Successful!");
-            invalidLoginCredentials.setStyle(successMessage);
-            loginUsernameTextField.setStyle(successStyle);
-            loginPasswordPasswordField.setStyle(successStyle);
-            invalidSignupCredentials.setText("");
+            invalid_login_credentials.setText("Login Successful!");
+            invalid_login_credentials.setStyle(successMessage);
+            login_username_text_field.setStyle(successStyle);
+            login_password_password_field.setStyle(successStyle);
+            invalid_signup_credentials.setText("");
         }
     }
 
     @FXML
     protected void onSignUpButtonClick() {
+        this.resetFields();
 
-        if (signUpUsernameTextField.getText().isBlank() || signUpEmailTextField.getText().isBlank() ||
-                signUpPasswordPasswordField.getText().isBlank() || signUpRepeatPasswordPasswordField.getText().isBlank()
+        boolean error = false;
+
+        if (sign_up_username_text_field.getText().isBlank() || sign_up_email_text_field.getText().isBlank() ||
+                sign_up_password_password_field.getText().isBlank() || sign_up_repeat_password_password_field.getText().isBlank()
         ) {
-            invalidSignupCredentials.setText("Please fill in all fields!");
-            invalidSignupCredentials.setStyle(errorMessage);
-            invalidLoginCredentials.setText("");
+            error = true;
+            invalid_signup_credentials.setText("Not all required fields are filled in.");
+            invalid_signup_credentials.setStyle(errorMessage);
 
-            if (signUpUsernameTextField.getText().isBlank()) signUpUsernameTextField.setStyle(errorStyle);
-            if (signUpEmailTextField.getText().isBlank()) signUpEmailTextField.setStyle(errorStyle);
-            if (signUpPasswordPasswordField.getText().isBlank()) signUpPasswordPasswordField.setStyle(errorStyle);
-            if (signUpRepeatPasswordPasswordField.getText().isBlank()) signUpRepeatPasswordPasswordField.setStyle(errorStyle);
+            if (sign_up_username_text_field.getText().isBlank())
+                sign_up_username_text_field.setStyle(errorStyle);
+            if (sign_up_email_text_field.getText().isBlank())
+                sign_up_email_text_field.setStyle(errorStyle);
+            if (sign_up_password_password_field.getText().isBlank())
+                sign_up_password_password_field.setStyle(errorStyle);
+            if (sign_up_repeat_password_password_field.getText().isBlank())
+                sign_up_repeat_password_password_field.setStyle(errorStyle);
```

Comment:
```suggestion
            if (sign_up_username_text_field.getText().isBlank())
                sign_up_username_text_field.setStyle(errorStyle);
                
            if (sign_up_email_text_field.getText().isBlank())
                sign_up_email_text_field.setStyle(errorStyle);
                
            if (sign_up_password_password_field.getText().isBlank())
                sign_up_password_password_field.setStyle(errorStyle);
                
            if (sign_up_repeat_password_password_field.getText().isBlank())
                sign_up_repeat_password_password_field.setStyle(errorStyle);
```

Format it this way everywhere, makes it look less cluttered.

Reply:
Fixed